src/transformers/data/data_tf_collator.py

```diff
@@ -0,0 +1,218 @@
+            raise ValueError(
+                "You are attempting to pad samples but the tokenizer you are using"
+                f" ({tokenizer.__class__.__name__}) does not have a pad token."
+            )
+
+        # Padding our tensor with the appropriate pad token.
+        max_length = length_of_first
+        if multiple_tensors:
+            for x in examples:
+                if x.shape[0] > max_length:
+                    max_length = tf.shape(x)[0]
+
+        if pad_to_multiple_of is not None and (max_length % pad_to_multiple_of != 0):
+            max_length = ((max_length // pad_to_multiple_of) + 1) * pad_to_multiple_of - length_of_first
+
+        token = tokenizer.pad_token_id if token_override is None else token_override
+
+        if tokenizer.padding_side == "right":
+            if multiple_tensors:
+                result = tf.pad(examples, [[0, 0], [0, max_length]], constant_values=token)
+            else:
+                result = tf.pad(examples, [[0, max_length]], constant_values=token)
+        else:
+            if multiple_tensors:
+                result = tf.pad(examples, [[0, 0], [max_length, 0]], constant_values=token)
+            else:
+                result = tf.pad(examples, [[max_length, 0]], constant_values=token)
+
+        return result
+
+    @tf.function()
+    def encode_objects(self, examples: Union[List[int], tf.Tensor, Dict[str, tf.Tensor]]) -> tf.data.Dataset:
+        if isinstance(examples, BatchEncoding):
+            examples = examples.data
+
+        if isinstance(examples, dict):
+            input = examples["input_ids"]
+        else:
+            input = examples
+
+        padded_output = self.tf_pad_tokens(input, self.tokenizer, self.padding_length)
+
+        encoded_batch = dict()
+        if isinstance(examples, dict):
+            encoded_batch.update(examples)
+            if self.padding_length is not None:
+                if "token_type_ids" in encoded_batch:
+                    encoded_batch["token_type_ids"] = self.tf_pad_tokens(
+                        encoded_batch["token_type_ids"], self.tokenizer, self.padding_length, 0
+                    )
+                if "attention_mask" in encoded_batch:
+                    encoded_batch["attention_mask"] = self.tf_pad_tokens(
+                        encoded_batch["attention_mask"], self.tokenizer, self.padding_length, 0
+                    )
+
+        # Mask example sequences and create their respective labels
+        encoded_batch["input_ids"], encoded_batch["labels"] = self.tf_mask_tokens(padded_output)
+        return encoded_batch
+
+    @tf.function
+    def square_ragged_tensors(self, examples):
+        if isinstance(examples, dict):
+            examples["input_ids"] = examples["input_ids"].to_tensor(0)
+            return examples
+        else:
+            return examples.to_tensor(0)
+
+    def __call__(self, examples: tf.data.Dataset, is_ragged=False) -> tf.data.Dataset:
```

Comment:
Whoops, I overlooked this bit! My intuition is that we should probably not be taking `tf.data.Dataset` as input, but instead writing this as a function we can call with either `dataset.map()` or with data as tensors or a dict of tensors. We don't want to lock people into `tf.data` if they're trying to write eager code, or doing their own data input.

Reply:
Good catch! Originally we wrote this as a function that took in a `tf.data.Dataset` due to the inclusion of ragged tensors. Now that we are planning on squaring off the tensors before inputting them, we can revisit a mapping approach.